Library/Homebrew/cmd/update.sh

```diff
@@ -637,13 +628,35 @@ EOS
     # the refspec ensures that the default upstream branch gets updated
     (
       UPSTREAM_REPOSITORY_URL="$(git config remote.origin.url)"
+      unset UPSTREAM_REPOSITORY
+      unset UPSTREAM_REPOSITORY_TOKEN
 
       # HOMEBREW_UPDATE_FORCE and HOMEBREW_UPDATE_AUTO aren't modified here so ignore subshell warning.
       # shellcheck disable=SC2030
       if [[ "${UPSTREAM_REPOSITORY_URL}" == "https://github.com/"* ]]
       then
         UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY_URL#https://github.com/}"
         UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY%.git}"
+      elif [[ "${UPSTREAM_REPOSITORY_URL}" =~ https://(([^:@]+)(:([^@]+))?@)?github.com/(.*)$ ]]
```

Comment:
```suggestion
      elif [[ "${UPSTREAM_REPOSITORY_URL}" =~ https://(([^:@/?#]+)(:([^@/?#]+))?@)github\.com/(.*)$ ]]
```

- Group 1 `?` seemed unnecessary as that just reduces it down to the earlier regex.
- `/` and similar URL component separators should not be matched in the username password as it's not valid there. Probably could be stricter but this is  good enough
- Escape `.` so we're only actually matching GitHub

Reply:
> * Group 1 `?` seemed unnecessary as that just reduces it down to the earlier `if` condition.

Is Group 1 itself even necessary? `${BASH_REMATCH[1]}` isn't used so perhaps remove group and shift numbers down.

Reply:
Looks not actually